src/Maestro/Microsoft.DotNet.Maestro.Tasks/src/PushMetadataToBuildAssetRegistry.cs

```diff
@@ -84,6 +84,7 @@ public async Task<bool> PushMetadataAsync(CancellationToken cancellationToken)
                     Client.Models.Build recordedBuild = await client.Builds.CreateAsync(finalBuild, cancellationToken);
 
                     Log.LogMessage(MessageImportance.High, $"Metadata has been pushed. Build id in the Build Asset Registry is '{recordedBuild.Id}'");
+                    Log.LogMessage(MessageImportance.High, $"##vso[task.setvariable variable=BARBuildId;]{recordedBuild.Id}");
```

Comment:
This should be a `Console.WriteLine` not a log, because the logs can be stopped. And this should not run unless we are inside azure pipelines.

Reply:
How to know if it's running inside AzP ?

Reply:
easiest way is to check if one of the environment variables is set. For a build check for something like "BUILD_BUILDID"

Reply:
Done